parasol_cpu/src/proc/fhe_processor.rs

```diff
@@ -468,14 +466,9 @@ impl FheProcessor {
     }
 
     pub fn retire(retirement_info: &RetirementInfo<DispatchIsaOp>, result: Result<()>) {
-        if let Err(e) = result {
-            // Waiting thread may have dropped.
-            let _ = retirement_info
-                .ready_instructions
-                .send(InstructionOperation::Retire(Err(e)));
-            return;
-        }
-
+        // We always notify our dependencies regardless of whether we errored or not.
+        // After doing that, we retire this instruction with success/failure as
+        // appropriate.
```

Comment:
We should set fault here rather than above so our dependencies will move to no-op mode rather than racing to go to no-op mode.